test/Microsoft.ML.Tests/DatabaseLoaderTests.cs

```diff
@@ -27,6 +27,20 @@ public DatabaseLoaderTests(ITestOutputHelper output)
 
         [LightGBMFact]
         public void IrisLightGbm()
+        {
+            DatabaseSource dbs = GetIrisDatabaseSource("SELECT * FROM {0}");
+            IrisLightGbmImpl(dbs);
+        }
+
+        [LightGBMFact]
+        public void IrisLightGbmWithTimeOut()
+        {
+            DatabaseSource dbs = GetIrisDatabaseSourceWithTimeOut("WAITFOR DELAY '00:00:01'; SELECT * FROM {0}", 1);
+            var ex = Assert.Throws<System.Reflection.TargetInvocationException>(() => IrisLightGbmImpl(dbs));
+            Assert.Contains("Timeout expired", ex.InnerException.Message);
```

Comment:
>Timeout expired [](start = 29, length = 15)

compare with ignorecase

Reply:
Is the exception thrown specific to a timeout? If so, can you compare the exact type of the exception?

---
In reply to: [451064274](https://github.com/dotnet/machinelearning/pull/5288#discussion_r451064274) [](ancestors = 451064274)

Reply:
> Is the exception thrown specific to a timeout? If so, can you compare the exact type of the exception?
> 
> In reply to: [451064274](https://github.com/dotnet/machinelearning/pull/5288#discussion_r451064274) [](ancestors = 451064274)

seems not...the exception type is TargetInvocationException(catches a SqlException) and its message contains "timeout"